community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ProcedureConfig.java

```diff
@@ -69,6 +73,17 @@ public ProcedureAllowedConfig( Config config )
                     } )
                     .collect( Collectors.toList() );
         }
+        String fullAccessProcedures =
+                config.getValue( Configuration.procedure_full_access.name() ).map( Object::toString ).orElse( "" );
+        if ( fullAccessProcedures.isEmpty() )
+        {
+            this.accessPatterns = Collections.emptyList();
+        }
+        else
+        {
+            this.accessPatterns = Stream.of( fullAccessProcedures.split( PROCEDURE_DELIMITER ) )
+                    .map( ProcedureConfig::compilePattern ).collect( Collectors.toList() );
+        }
```

Comment:
This new functionality and parsing is very similar to the `allowedRoles`. Maybe the parsing and compilation of procedure matching regex patterns can be unified?